packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/sync/SyncConfiguration.kt

```diff
@@ -63,13 +89,28 @@ public interface SyncConfiguration : Configuration {
     // public val partitionValue: PartitionValue
     public val errorHandler: SyncSession.ErrorHandler?
 
+    /**
+     * The mode of synchronization for this Realm
+     */
+    public val syncMode: SyncMode
+
+    // /**
+    //  * TODO https://github.com/realm/realm-kotlin/issues/840
+    //  */
+    // public val initialSubscriptionsCallback: InitialSubscriptionsCallback?
+    //
+    // /**
+    //  * TODO https://github.com/realm/realm-kotlin/issues/840
+    //  */
+    // public val rerunInitialSubscriptions: Boolean
+
     /**
      * Used to create a [SyncConfiguration]. For common use cases, a [SyncConfiguration] can be
      * created using the [SyncConfiguration.with] function.
      */
     public class Builder private constructor(
         private var user: User,
-        private var partitionValue: PartitionValue,
+        private var partitionValue: PartitionValue?,
```

Comment:
We might like to be more explicit to the users what type of Sync config they are choosing, and rather leaving it to the signature of the constructor we could define it at type level, like having a `FlexibleSyncConfiguration` and `PartitionSyncConfiguration`.

Reply:
We talked a bit about this when adding Flexible Sync to Java... IMO doing that is just extra work for no real benefit, i.e.:

- You still only have a Realm instance which can be issued, i.e. `realm.subscriptions` will be available regardless of the SyncConfiguration type.

- Partition-based Sync is going away soon anyway, so having a `FlexibleSyncConfiguration` would be annoying there.

- Also, Claus had another similar comment elsewhere, so I refactored the API slightly so there now is a `SyncMode` enum and a `SyncConfiguration.syncMode` that can either return `SyncMode.PARTION_BASED` or `SyncMode.FLEXIBLE`.